comment_scraper.py

```diff
@@ -1,87 +1,96 @@
+            build_comments_list(client, asset['id'], comment_list)
+
+        if asset['type'] == "file" and asset['comment_count'] > 0:
+            comments = client.get_comments(asset['id'])
+            for comment in comments:
+                # The 'get_comments" call won't return the asset name or parent ID
+                # So we'll add them to the dictionary now. 
+                comment['parent_id'] = asset['parent_id']
+                comment['name'] = asset['name']
+                comment_list.append(comment)
 
         if asset['type'] == "version_stack":
+            # Read about version stacks: https://docs.frame.io/docs/managing-version-stacks
+            versions = client.get_asset_children(asset['id'])
+            for v_asset in versions:
+                comments = client.get_comments(v_asset['id'])
+                for comment in comments:
+                    comment['parent_id'] = v_asset['parent_id']
+                    comment['name'] = v_asset['name']
+                    comment_list.append(comment)
 
-# Saving the asset name and parent ID.
+    return comment_list
 
-            asset_name = asset['name']
-            parent_id = asset['parent_id']
-            vfiles = client.get_asset_children(asset['id'])
 
-            for asset in vfiles.results:
-                asset_name = asset['name']
-                parent_id = asset['parent_id']
-                if asset['type'] == "file":
-                    if asset['comment_count'] > 0:
-                        comments = client.get_comments(asset['id'])
-                        my_comment_list = [comment for comment in comments.results]
-                        for object in my_comment_list:
-                            object.update({'parent_id':parent_id})
-                            object.update({'name':asset_name})
-                        comment_list.append(my_comment_list)
+def flatten_dict(d):
+    # The get_comments API response is verbose and contains nested objects.
+    # Use this helper functon to flatten the dict holding the API response data
+    # and namespace the attributes.
 
-# Takes a root asset ID for a project and a developer token.
-# Returns a comment list with all assets.
+    def expand(key, val):
+            if isinstance(val, dict):
+                return [ (key + '.' + k, v) for k, v in flatten_dict(val).items() ]
+            else:
+                return [ (key, val) ]
+    
+    items = [ item for k, v in d.items() for item in expand(k, v)]
 
-def get_all_project_comments(root_asset_id, token):
-    comment_list = []
-    client = FrameioClient(token)
+    return dict(items)
 
-    all_comments(client, root_asset_id, comment_list)
 
-    return comment_list
+def write_comments_csv(c_list):
+    # Writes comments to comments.csv
+    # Any attributes you add to the headers list will automatically be written to the CSV
+    # The API returns many attributes so familiarize with the response data!
```

Comment:
```suggestion
    # The API returns many attributes so familiarize yourself with the response data!
```
is there a link you could include here that shows what all of them are?